### PR TITLE
add a KIP-7 API 

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

- This PR added a KIP-7 API
  - added a `KIP7` class and `KIP7QueryOptions` class.
  - added KIP-7 rest-client code.
  - added shell script that generates KIP-7 rest-client codes. 

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/ground-x/caver-java-ext-kas/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/ground-x/caver-java-ext-kas)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 1]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
     }
 
     /**
-     * Getter function for Token History
-     * @return TokenHistoryAPI.
+     * Getter function for Token History.
+     * @return TokenHistoryAPI
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:167]:
```suggestion
     * Getter function for TokenHistory API instance.
     * @return TokenHistory
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 2]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
...
      */
     public TokenHistory getTokenHistory() {
         return tokenHistory;
     }
 
     /**
      * Setter function for Token History.
-     * @param tokenHistory The Token History API Instance
+     * @param tokenHistory The Token History API Instance.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:175]:
```suggestion
     * @param tokenHistory The TokenHistory API instance.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 3]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
...
      */
     public void setTokenHistory(TokenHistory tokenHistory) {
         this.tokenHistory = tokenHistory;
     }
 
     /**
-     * Getter function for wallet
+     * Getter function for wallet.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:182]:
```suggestion
     * Getter function for Wallet API instance.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 4]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
...
      * @return WalletAPI
      */
     public Wallet getWallet() {
         return wallet;
     }
 
     /**
-     * Setter function for Wallet
+     * Setter function for Wallet.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:190]:
```suggestion
     * Setter function for Wallet API instance.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 5]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
...
      * @param wallet The WalletAPI instance.
      */
     public void setWallet(Wallet wallet) {
         this.wallet = wallet;
     }
 
     /**
-     * Getter function for kip17
+     * Getter function for kip17.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:198]:
```suggestion
     * Getter function for KIP17 API instance.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java [hunk 6]
```diff
@@ -139,50 +163,66 @@ public void setAnchor(Anchor anchor) {
...
+     * @param tokenHistory The Token History API Instance.
      */
     public void setTokenHistory(TokenHistory tokenHistory) {
         this.tokenHistory = tokenHistory;
     }
 
     /**
-     * Getter function for wallet
+     * Getter function for wallet.
      * @return WalletAPI
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/KAS.java:183]:
```suggestion
     * @return Wallet
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 7]
```diff
@@ -0,0 +1,1459 @@
+package xyz.groundx.caver_ext_kas.kas.kip7;
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:1]: Can you add license?

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 8]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Kip7ContractMetadataResponse
+     * @throws ApiException
+     */
+    public Kip7ContractMetadataResponse getContract(String addressOrAlias) throws ApiException {
+        return kip7Api.getContract(chainId, addressOrAlias);
+    }
+
+    /**
+     * Retrieves KIP-7 contract information by either contract address or alias. <br>
+     * GET /v1/contract/{contract-address-or-alias}
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:228]:
```suggestion
     * GET /v1/contract/{contract-address-or-alias} <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 9]
```diff
@@ -0,0 +1,1459 @@
...
+import xyz.groundx.caver_ext_kas.rest_client.io.swagger.client.ApiClient;
+import xyz.groundx.caver_ext_kas.rest_client.io.swagger.client.ApiException;
+import xyz.groundx.caver_ext_kas.rest_client.io.swagger.client.api.kip7.api.Kip7Api;
+import xyz.groundx.caver_ext_kas.rest_client.io.swagger.client.api.kip7.api.Kip7DeployerApi;
+import xyz.groundx.caver_ext_kas.rest_client.io.swagger.client.api.kip7.model.*;
+
+import java.math.BigInteger;
+
+/**
+ * Representing an wrapping class that connects KIP-7 APi.
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:15]:
```suggestion
 * Representing an wrapping class that connects KIP-7 API.
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 10]
```diff
@@ -0,0 +1,1459 @@
...
+        request.setOwner(owner);
+        request.setSpender(spender);
+        request.setAmount(amount);
+
+        return kip7Api.approveTokenAsync(chainId, addressOrAlias, request, callback);
+    }
+
+    /**
+     * Sends tokens of the contract. <br>
+     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:562]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 11]
```diff
@@ -0,0 +1,1459 @@
...
+     * <pre>Example :
+     * {@code
+     * String contractAlias = "";
+     * String owner = "";
+     * String to = "";
+     * BigInteger amount = BigInteger.valueOf(10).multiply(BigInteger.TEN.pow(18)); // 10 * 10^18
+     *
+     * Kip7TransactionStatusResponse response = caver.kas.kip7.transfer(contractAlias, owner, spender, amount);
+     * }
+     * </pre>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:572]:
```suggestion
     * </pre>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 12]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Kip7TransactionStatusResponse
+     * @throws ApiException
+     */
+    public Kip7TransactionStatusResponse transfer(String addressOrAlias, String from, String to, BigInteger amount) throws ApiException {
+        return transfer(addressOrAlias, from, to, Numeric.toHexStringWithPrefix(amount));
+    }
+
+    /**
+     * Sends tokens of the contract asynchronously. <br>
+     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:586]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 13]
```diff
@@ -0,0 +1,1459 @@
...
+     *      ....implements callback method.
+     * }
+     *
+     * String testAlias = "test-contract";
+     * String name = "TEST_KIP7";
+     * String symbol = "TKIP7";
+     * int decimals = 18;
+     * BigInteger initial_supply = BigInteger.valueOf(100_000).multiply(BigInteger.TEN.pow(18)); // 100000 * 10^18
+     *
+     * caver.kas.kip7.deployAsync(name, symbol, decimals, initial_supply, testAlias, callback);
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:96]:
```suggestion
     * BigInteger initialSupply = BigInteger.valueOf(100_000).multiply(BigInteger.TEN.pow(18)); // 100000 * 10^18
     *
     * caver.kas.kip7.deployAsync(name, symbol, decimals, initialSupply, testAlias, callback);
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 14]
```diff
@@ -0,0 +1,1459 @@
...
+     * }
+     *
+     * String contractAlias = "";
+     * String owner = "";
+     * String to = "";
+     * BigInteger amount = BigInteger.valueOf(10).multiply(BigInteger.TEN.pow(18)); // 10 * 10^18
+     *
+     * caver.kas.kip7.transferAsync(contractAlias, owner, to, amount, callback);
+     * }
+     * </pre>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:600]:
```suggestion
     * </pre>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 15]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Call
+     * @throws ApiException
+     */
+    public Call transferAsync(String addressOrAlias, String from, String to, BigInteger amount, ApiCallback<Kip7TransactionStatusResponse> callback) throws ApiException {
+        return transferAsync(addressOrAlias, from, to, Numeric.toHexStringWithPrefix(amount), callback);
+    }
+
+    /**
+     * Sends tokens of the contract. <br>
+     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:615]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 16]
```diff
@@ -0,0 +1,1459 @@
...
+     * <pre>Example :
+     * {@code
+     * String contractAlias = "";
+     * String owner = "";
+     * String to = "";
+     * String amount = "0x8ac7230489e80000"; // 10 * 10^18
+     *
+     * Kip7TransactionStatusResponse response = caver.kas.kip7.transfer(contractAlias, owner, spender, amount);
+     * }
+     * </pre>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:625]:
```suggestion
     * </pre>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 17]
```diff
@@ -0,0 +1,1459 @@
...
+     *
+     * <pre>Example :
+     * {@code
+     * String testAlias = "test-contract";
+     * String name = "TEST_KIP7";
+     * String symbol = "TKIP7";
+     * int decimals = 18;
+     * BigInteger initial_supply = BigInteger.valueOf(100_000).multiply(BigInteger.TEN.pow(18)); // 100000 * 10^18
+     *
+     * Kip7TransactionStatusResponse response = caver.kas.kip7.deploy(name, symbol, decimals, initial_supply, testAlias);
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:64]:
```suggestion
     * BigInteger initialSupply = BigInteger.valueOf(100_000).multiply(BigInteger.TEN.pow(18)); // 100000 * 10^18
     *
     * Kip7TransactionStatusResponse response = caver.kas.kip7.deploy(name, symbol, decimals, initialSupply, testAlias);
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 18]
```diff
@@ -0,0 +1,1459 @@
...
+        request.setFrom(from);
+        request.setTo(to);
+        request.setAmount(amount);
+
+        return kip7Api.transferToken(chainId, addressOrAlias, request);
+    }
+
+    /**
+     * Sends tokens of the contract asynchronously. <br>
+     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:644]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/transfer <br>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 19]
```diff
@@ -0,0 +1,1459 @@
...
+     * }
+     *
+     * String contractAlias = "";
+     * String owner = "";
+     * String to = "";
+     * String amount = "0x8ac7230489e80000"; // 10 * 10^18
+     *
+     * caver.kas.kip7.transferAsync(contractAlias, owner, to, amount, callback);
+     * }
+     * </pre>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:658]:
```suggestion
     * </pre>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 20]
```diff
@@ -0,0 +1,1459 @@
...
+     *
+     * <pre>Example :
+     * {@code
+     * String testAlias = "test-contract";
+     * String name = "TEST_KIP7";
+     * String symbol = "TKIP7";
+     * int decimals = 18;
+     * String initial_supply = "0x152d02c7e14af6800000";
+     *
+     * Kip7TransactionStatusResponse response = caver.kas.kip7.deploy(name, symbol, decimals, initial_supply, testAlias);
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:125]:
```suggestion
     * String initialSupply = "0x152d02c7e14af6800000";
     *
     * Kip7TransactionStatusResponse response = caver.kas.kip7.deploy(name, symbol, decimals, initialSupply, testAlias);
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 21]
```diff
@@ -0,0 +1,1459 @@
...
+        request.setFrom(from);
+        request.setTo(to);
+        request.setAmount(amount);
+
+        return kip7Api.transferTokenAsync(chainId, addressOrAlias, request, callback);
+    }
+
+    /**
+     * Sends tokens on behalf of the owner. <br>
+     * It need to approve token transfer in advance {@link #approve(String, String, String, String)} approve} method.
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:678]:
```suggestion
     * It need to approve token transfer in advance {@link #approve(String, String, String, String)} approve} method. <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 22]
```diff
@@ -0,0 +1,1459 @@
...
+     * POST /v1/contract/{contract-address-or-alias}/transfer-from
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:679]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/transfer-from <br>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 23]
```diff
@@ -0,0 +1,1459 @@
...
+     * String contractAlias = "";
+     * String owner = "";
+     * String spender = "";
+     * String to = "";
+     * BigInteger amount = BigInteger.valueOf(10).multiply(BigInteger.TEN.pow(18)); // 10 * 10^18
+     *
+     * Kip7TransactionStatusResponse transferRes = caver.kas.kip7.transferFrom(contractAlias, spender, owner, to, amount);
+
+     * }
+     * </pre>
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:691]:
```suggestion
     * </pre>
     
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 24]
```diff
@@ -0,0 +1,1459 @@
...
+        MintKip7TokenRequest request = new MintKip7TokenRequest();
+        request.setTo(to);
+        request.setAmount(amount);
+
+        return kip7Api.mintTokenAsync(chainId, addressOrAlias, request, callback);
+    }
+
+    /**
+     * Burns tokens. <br>
+     * POST /v1/contract/{contract-address-or-alias}/burn
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:929]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/burn <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 25]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Kip7TransactionStatusResponse
+     * @throws ApiException
+     */
+    public Kip7TransactionStatusResponse burn(String addressOrAlias, BigInteger amount) throws ApiException {
+        return burn(addressOrAlias, null, Numeric.toHexStringWithPrefix(amount));
+    }
+
+    /**
+     * Burns tokens asynchronously. <br>
+     * POST /v1/contract/{contract-address-or-alias}/burn
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:951]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/burn <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 26]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Call
+     * @throws ApiException
+     */
+    public Call burnAsync(String addressOrAlias, BigInteger amount, ApiCallback<Kip7TransactionStatusResponse> callback) throws ApiException {
+        return burnAsync(addressOrAlias, null, amount, callback);
+    }
+
+    /**
+     * Burns tokens. <br>
+     * POST /v1/contract/{contract-address-or-alias}/burn
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:978]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/burn <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 27]
```diff
@@ -0,0 +1,1459 @@
...
+     * @return Kip7TransactionStatusResponse
+     * @throws ApiException
+     */
+    public Kip7TransactionStatusResponse burn(String addressOrAlias, String amount) throws ApiException {
+        return burn(addressOrAlias, null, amount);
+    }
+
+    /**
+     * Burns tokens asynchronously. <br>
+     * POST /v1/contract/{contract-address-or-alias}/burn
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:1000]:
```suggestion
     * POST /v1/contract/{contract-address-or-alias}/burn <br>
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 28]
```diff
@@ -0,0 +1,1459 @@
...
+     * <pre>Example :
+     * {@code
+     * Kip7ContractListResponse response = caver.kas.kip7.getContractList();
+     * }
+     * </pre>
+     *
+     * @return Kip7ContractListResponse
+     * @throws ApiException
+     */
+    public Kip7ContractListResponse getContractList() throws ApiException {
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:202]: Can you adjust location to put `getContractList` functions together?

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 29]
```diff
@@ -0,0 +1,1459 @@
...
+        return getContractList(new KIP7QueryOptions());
+    }
+
+    /**
+     * Retrieves KIP-7 contract information by either contract address or alias. <br>
+     * GET /v1/contract/{contract-address-or-alias}
+     *
+     * <pre>Example :
+     * {@code
+     * String contractAlias = "";
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:212]:
```suggestion
     * String contractAlias = "{contract alias string}";
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java (hunk 24 above)
Comment (aeharvlee) [reply src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:929]: @sirano11 
Hi sirano. I think we should make a convention about it.
Some lines have all the <br> tags, and some not on the line.

It seems good to set up a convention about it and follow it.

I will not leave suggestions for corrections in the rest of the similar parts.
We should think about it first I think 😃

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 30]
```diff
@@ -0,0 +1,1459 @@
...
+     *       ....implements callback method.
+     * }
+     *
+     * String contractAlias = "";
+     * caver.kas.kip7.getContractAsync(contractAlias, callback);
+     * }
+     * </pre>
+     *
+     * @param addressOrAlias Contract address (in hexadecimal with the 0x prefix) or an alias.
+     * @param addressOrAlias
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:242]:
```suggestion
```

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 31]
```diff
@@ -0,0 +1,1459 @@
...
+     * <pre>Example :
+     * {@code
+     * ApiCallback<Kip7TokenBalanceResponse> callback = new ApiCallback<Kip7TransactionStatusResponse>() {
+     *     ....implements callback method.
+     * }
+     *
+     * String contractAlias = "";
+     * String owner = "";
+     *
+     * caver.kas.kip7.balanceAsync(contractAlias, owner, callback);
```
Comment (aeharvlee) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:422]:
```suggestion
     * caver.kas.kip7.balanceAsync(contractAlias, owner, callback);
     }
```
There was a javadoc build error. :)

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 32]
```diff
@@ -0,0 +1,1459 @@
...
+     * </pre>
+     *
+     * @param addressOrAlias Contract address (in hexadecimal with the 0x prefix) or an alias.
+     * @param owner Klaytn account address of the owner delegating the token tranfer. The default is the contract deployer account.
+     * @param spender Klaytn account address to approve delegated token transfer.
+     * @param amount Approved token amount. (in hexadecimal with the 0x prefix)
+     * @return Kip7TransactionStatusResponse
+     * @throws ApiException
+     */
+    public Kip7TransactionStatusResponse approve(String addressOrAlias, String owner, String spender, String amount) throws ApiException {
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:515]: Can you implement `public Kip7TransactionStatusResponse approve(String addressOrAlias, String spender, String amount)` function which does not receive `owner` parameter? The default is the contract deployer account (In yaml file, `ApproveKip7TokenRequest` component has info for this).

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 33]
```diff
@@ -0,0 +1,1459 @@
...
+     * }
+     * </pre>
+     * @param addressOrAlias Contract address (in hexadecimal with the 0x prefix) or an alias.
+     * @param from Klaytn account address to send tokens.
+     * @param to Klaytn account addree to receive tokens
+     * @param amount The amount of tokens to transfer. (in hexadecimal with the 0x prefix)
+     * @return Kip7TransactionStatusResponse
+     * @throws ApiException
+     */
+    public Kip7TransactionStatusResponse transfer(String addressOrAlias, String from, String to, String amount) throws ApiException {
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:649]: The `from` parameter here also can be skipped. Please check `TransferKip7TokenRequest` component. The default value of the `from` is the contract deployer account.

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java (hunk 7 above)
Comment (sirano11) [reply src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:17]: I added it.

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java (hunk 33 above)
Comment (sirano11) [reply src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:745]: I added it.

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java (hunk 32 above)
Comment (sirano11) [reply src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:639]: I added it.

File: src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java [hunk 34]
```diff
@@ -0,0 +1,1693 @@
...
+        return kip7Api.getTokenAllowanceAsync(chainId, addressOrAlias, owner, spender, callback);
+    }
+
+    /**
+     * Shows the `{owner}`'s balance for the given KIP-7 contract. <br>
+     * GET /v1/contract/{contract-address-or-alias}/account/{owner}/balance
+     *
+     * <pre>Example :
+     * {@code
+     * String contractAddress = "";
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/kas/kip7/KIP7.java:409]:
```suggestion
     * String contractAddress = "{contract address}";
```

**Review by jimni1222 — APPROVED** [review]
LGTM =) Thank you for your hard working!

**Review by aeharvlee — APPROVED** [review]

**Pull request merged**